rdkit/Chem/Fraggle/FraggleSim.py

```diff
@@ -29,356 +29,285 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #
 # Created by Jameed Hussain, May 2013
+"""
+Fragmentation algorithm
+-----------------------
+
+identify acyclic bonds
+enumerate all single cuts
+make sure you chop off more that 1 atom
+keeps bits which are >60% query mol
+enumerate all double cuts
+keeps bits with 1 attachment point (i.e throw middle bit away)
+need to be >60% query mol
+
+identify exocyclic bonds
+enumerate all single "ring" cuts
+Check if it results in more that one component
+keep correct bit if >40% query mol
+
+enumerate successful "rings" cuts with an acyclic cut
+Check if it results in more that one component
+keep correct if >60% query mol
+
+"""
 import sys
 from rdkit import Chem, DataStructs
+from rdkit import rdBase
+from itertools import combinations
 
 # our default rdkit fingerprinter parameters:
-rdkitFpParams = dict(maxPath=5, fpSize=1024, nBitsPerHash=2)
```

Comment:
Curious why you made this change.
Both forms are legit Python and I find the first more readable. 
I think I remember reading about a performance difference, but that's hardly relevant here.